frontend/app/src/MetricsManager.ts

```diff
@@ -78,7 +88,13 @@ export class MetricsManager {
     gatherUsageStats: boolean
   }): void {
     this.initialized = true
-    this.actuallySendMetrics = gatherUsageStats
+    // Handle if the user or the host has disabled metrics
+    this.actuallySendMetrics = gatherUsageStats && this.metricsUrl !== "off"
+
+    // Trigger fallback to fetch default metrics config if not provided by host
+    if (this.actuallySendMetrics && !this.metricsUrl) {
+      this.requestDefaultMetricsConfig()
+    }
```

Comment:
nit: We could set `actuallySendMetrics` to False here in case that `this.metricsUrl` is undefined after the `requestDefaultMetricsConfig` to ensure that metrics gathering gets deactivated. But it depends a bit on the full implementation if this is actually required.